File: Web/src/data/models/CreditCardRegistry.ts
import { Registry, RegistryType } from "./Registry";

export interface WithInvoiceTime {
  month: number;
  year: number;
}

export class CreditCardRegistry extends Registry implements WithInvoiceTime {
  constructor(
    id: string,
    public cardId: string,
    public month: number, public year: number, date: Date,
    description: string,
    value: number,
    tags: string[] = [],
    categoryId?: string,
    observation?: string,
    importInfo?: any
  ) { 
    super(id, RegistryType.CREDIT, true, value, description, date, tags, categoryId, observation, importInfo);
  }
}

File: iOS/Sources/Data/models/CreditCardRegistry.swift
import Foundation

struct CreditCardRegistry: Codable, Identifiable {
    var id: String?
    var type: RegistryType = .credit
    var cardId: String
    var invoiceId: String?
    var value: Double
    var description: String
    var date: Date
    var paid: Bool
    var tags: [String]
    var categoryId: String?
    var observation: String?
    var relatedInfo: String?
}
